src/Cli/dotnet/SlnFileExtensions.cs

```diff
@@ -93,6 +95,11 @@ public static void AddProject(this SlnFile slnFile, string fullProjectPath, ILis
             }
         }
 
+        private static bool AreBuildConfigurationsApplicable(this SlnFile slnFile, string path)
+        {
+            return !Path.GetExtension(path).Equals(".shproj", StringComparison.OrdinalIgnoreCase);
+        }
+
```

Comment:
Could remove the `slnFile` parameter.

Reply:
AreBuildConfigurationsApplicable does not use its `slnFile` parameter, but AddDefaultBuildConfigurations does: <https://github.com/dotnet/sdk/blob/30ebf8a2407da49bb93acb21b422dc52d6ddc2e1/src/Cli/dotnet/SlnFileExtensions.cs#L105>

SlnFileExtensions also has other private methods that do not have any `slnFile` parameter: <https://github.com/dotnet/sdk/blob/30ebf8a2407da49bb93acb21b422dc52d6ddc2e1/src/Cli/dotnet/SlnFileExtensions.cs#L176>

Reply:
True, I've updated the code accordingly.